homeassistant/components/tradfri/base_class.py

```diff
@@ -84,8 +94,10 @@ class TradfriBaseDevice(TradfriBaseClass):
     """
 
     @property
-    def device_info(self):
+    def device_info(self) -> DeviceInfo:
         """Return the device info."""
+        if not self._device:
+            return {}
```

Comment:
Method claims to return `DeviceInfo`, but returns dict here.

Reply:
yes, empty deviceInfo, which seems correct when the device is not present.